src/Controller/HistoryController.php

```diff
@@ -105,4 +108,105 @@ public function isMediaProcessing(): JsonResponse
 
         return new JsonResponse($result);
     }
+
+    #[Route(
+        path: '/api/_action/migration/get-log-groups',
+        name: 'api.admin.migration.get-log-groups',
+        methods: ['GET'],
+        defaults: ['_acl' => ['swag_migration.viewer']]
+    )]
+    public function getLogGroups(Request $request, Context $context): JsonResponse
+    {
+        $runId = $request->query->getAlnum('runId');
+        $level = $request->query->get('level', '');
+        $pageParam = $request->query->get('page');
+        $limitParam = $request->query->get('limit');
+
+        $sortBy = $request->query->get('sortBy', 'count');
+        $sortDirection = $request->query->get('sortDirection', 'DESC');
+
+        $filterCode = $request->query->get('filterCode');
+        $filterStatus = $request->query->get('filterStatus');
+        $filterEntity = $request->query->get('filterEntity');
+        $filterField = $request->query->get('filterField');
+
+        if (empty($runId) || empty($level)) {
+            throw RoutingException::missingRequestParameter($runId === '' ? 'runId' : 'level');
+        }
+
+        if (!\is_numeric($pageParam)) {
+            throw RoutingException::invalidRequestParameter('page');
+        }
+
+        if (!\is_numeric($limitParam)) {
+            throw RoutingException::invalidRequestParameter('limit');
+        }
+
+        if (!\is_string($sortBy) || empty($sortBy)) {
+            $sortBy = 'count';
+        }
+
+        if (!\is_string($sortDirection) || !\in_array(\strtoupper($sortDirection), ['ASC', 'DESC'], true)) {
+            $sortDirection = 'DESC';
+        }
+
+        $result = $this->logGroupingService->getGroupedLogsByCodeAndEntity(
+            $runId,
+            $level,
+            (int) $pageParam,
+            (int) $limitParam,
```

Comment:
~~I think there should be a better fallback than just casting to `int`, when these query params aren't available~~

Nevermind, you already throw a exception in these cases. But are these params really required?